site/formulas/decoding.md

```diff
@@ -1 +1,39 @@
 # Decoding to Extended Coordinates
+
+Given 32 octets of data, `s_bytes`, representing a compressed
+Ristretto point, the implementation must:
+
+1. Check that `s_bytes` is in fact 32 octets.  (In typed languages the
+   implementor should guarantee this through the type system, otherwise
+   there should be an explicit check.)
+2. Decode `s_bytes` into a field element, `s`, then re-encode `s` into
+   another array of 32 octets, `s_bytes_check`.  Check that the two
+   arrays `s_bytes` and `s_bytes_check` are equal in constant time.
+   This is done to ensure that the field element `s` satisfies
+   \\( s < p : p \in (2^{255}, 2^{255}-19] \\) (for the case of
```

Comment:
This description describes how we implemented "check that the encoding is canonical", but it doesn't describe that that's the goal of what we're doing --- and while the checks we do in our implementation do accomplish that goal, they work because of other properties that might not be true of other implementations.

For this reason I think it would be better to instead say "check that the encoding is canonical", and then have a note that one way to do that is by decoding, computing the canonical encoding, and checking that that was the original input. But for this to work, the field element encoding function has to be canonical, which it might or might not be.

Reply:
See 9c06283